pkg/registry/core/resourcequota/strategy.go

```diff
@@ -46,13 +47,15 @@ func (resourcequotaStrategy) NamespaceScoped() bool {
 func (resourcequotaStrategy) PrepareForCreate(ctx context.Context, obj runtime.Object) {
 	resourcequota := obj.(*api.ResourceQuota)
 	resourcequota.Status = api.ResourceQuotaStatus{}
+	resourcequotautil.DropDisabledFields(&resourcequota.Spec, nil)
```

Comment:
the reason why pods have this function as a library is because they are reused (for example, for PodSpecs within Depoyments or other workload APIs).

There is no really a need to reuse this function, so you can just implement in this file.

Reply:
The reason I did this is because I looked at the history of how feature gates were applied to resource quota, and this was the practice even when there was a single field to check.